src/cljx/schema/core.cljx

```diff
@@ -573,6 +573,26 @@
       (macros/error! (utils/format* "Not an IDeref: %s" schema)))
     (Recursive. schema)))
 
+#+cljs
+(do
+  (cljs.core/defrecord Recursive [derefable]
+    Schema
+    (walker [this]
+      (let [a (atom nil)]
+        (reset! a (start-walker
+                   (let [old subschema-walker]
+                     (clojure.core/fn [s] (if (= s this) #(@a %) (old s))))
+                   @derefable))))
+    (explain [this]
+      (let [{:keys [ns name]} (meta derefable)]
+        (list 'recursive (str ns "/" name)))))
```

Comment:
Good observation! I was focused on just making it work at first :). Will try to grab time to look at code sharing.